modin/core/storage_formats/pandas/utils.py

```diff
@@ -90,12 +90,11 @@ def split_result_of_axis_func_pandas(axis, num_splits, result, length_list=None)
     if num_splits == 1:
         return [result]
 
-    if length_list is not None:
-        length_list.insert(0, 0)
-    else:
-        chunksize = compute_chunksize(result.shape[axis], num_splits)
-        length_list = np.full(num_splits + 1, chunksize)
-        length_list[0] = 0
+    if length_list is None:
+        length_list = get_length_list(result.shape[axis], num_splits)
```

Comment:
moved the logic of getting axis lengths into a separate function to reuse it in the cache computation

Reply:
New function works a little differently from how it was before. For example, when len == 1001 and num_splits == 8. Old array: `array([  0, 126, 126, 126, 126, 126, 126, 126, 126])` vs new array: `array([  0, 126, 126, 126, 126, 126, 126, 126, 119])`. It does not affect the following code?

Also, your function creates an intermediate list, which looks slower. Is this slowdown insignificant?

Reply:
> New function works a little differently from how it was before

It doesn't really make a difference in the end. The sum of the "old" vs "new" array is 1008 vs 1001, meaning that the new function just additionally checks for axis bounds. It wasn't required before as these lengths were only used to split the frame via `.iloc` which doesn't fail on out-of-bounds slices, but since now we use these lengths as the actual partitions lengths/widths it's mandatory for them to not be out of bounds.

> Also, your function creates an intermediate list, which looks slower. Is this slowdown insignificant?

Running `split_result_of_axis_func_pandas` 1_000 times gives the following cumulative number of seconds:
|master|this PR|
|-|-|
|10.12s|10.44s|

So yeah, it's a bit slower, but I don't think it's significant as given this number the difference for a single call is 0.0003 seconds.


<details><summary>Testing code</summary>

```python
from timeit import timeit
from modin.core.storage_formats.pandas.utils import split_result_of_axis_func_pandas
import pandas

NROWS = 10_000
NCOLS = 64
NSPLITS = 1024

df = pandas.DataFrame({f"col{i}": range(NROWS) for i in range(NCOLS)})
print(
    timeit(
        lambda: split_result_of_axis_func_pandas(axis=0, num_splits=NSPLITS, result=df),
        number=1_000,
    )
)
```

</details>

Reply:
> > New function works a little differently from how it was before
> 
> It doesn't really make a difference in the end. The sum of the "old" vs "new" array is 1008 vs 1001, meaning that the new function just additionally checks for axis bounds. It wasn't required before as these lengths were only used to split the frame via `.iloc` which doesn't fail on out-of-bounds slices, but since now we use these lengths as the actual partitions lengths/widths it's mandatory for them to not be out of bounds.
> 
> > Also, your function creates an intermediate list, which looks slower. Is this slowdown insignificant?
> 
> Running `split_result_of_axis_func_pandas` 1_000 times gives the following cumulative number of seconds:
> 
> master	this PR
> 10.12s	10.44s
> So yeah, it's a bit slower, but I don't think it's significant as given this number the difference for a single call is 0.0003 seconds.
> 
> Testing code
> ```python
> from timeit import timeit
> from modin.core.storage_formats.pandas.utils import split_result_of_axis_func_pandas
> import pandas
> 
> NROWS = 10_000
> NCOLS = 64
> NSPLITS = 1024
> 
> df = pandas.DataFrame({f"col{i}": range(NROWS) for i in range(NCOLS)})
> print(
>     timeit(
>         lambda: split_result_of_axis_func_pandas(axis=0, num_splits=NSPLITS, result=df),
>         number=1_000,
>     )
> )
> ```

Good, thanks.